extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AbstractDataSourceProducer.java

```diff
@@ -201,6 +202,18 @@ public boolean isValid(Connection connection) {
             poolConfiguration.maxLifetime(dataSourceRuntimeConfig.maxLifetime.get());
         }
 
+        if (dataSourceRuntimeConfig.socketFactory.isPresent()) {
+            agroalConnectionFactoryConfigurationSupplier.jdbcProperty("socketFactory",
```

Comment:
Since `socketFactory` is a JDBC property, the best is to have it declared in the `additionalJdbcProperties`, like in:

`quarkus.datasource.additional-jdbc-properties.socketFactory=foobar`